src/fsd/4-entities/upgrade/upgrades.service.ts

```
@@ -23,6 +26,178 @@ export class UpgradesService {
+            stat: upgrade.stat,
+        };
+        let moreToExpand = false;
+        for (const recipeItem of upgrade.recipe) {
+            if (!expandedRecipeData[recipeItem.id]) {
+                // We haven't expanded an ingredient yet, so we can't expand this recipe.
+                moreToExpand = true;
+                break;
+            }
+            if (!expandedRecipeData[recipeItem.id].crafted) {
+                // Simple ingredient, just add it.
+                this.addIngredientsToExpandedRecipe(expandedRecipe, {
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)